x-pack/filebeat/input/s3/s3_integration_test.go

```diff
@@ -128,7 +151,7 @@ func collectOldMessages(t *testing.T, queueURL string, visibilityTimeout int64,
 	return oldMessageHandles
 }
 
-func (input *s3Input) deleteAllMessages(t *testing.T, awsConfig awssdk.Config, queueURL string, visibilityTimeout int64, svcSQS sqsiface.ClientAPI) error {
+func (input *s3Collector) deleteAllMessages(t *testing.T, queueURL string, visibilityTimeout int64, svcSQS sqsiface.ClientAPI) error {
```

Comment:
Let's not add methods for testing to existing objects. If we need some utilities either provide functions, or wrap the original object and provide a more extended testing 'experience. I think in this case a simple function might suffice.

When do we delete entries? On beginning or at end of test?

Why do we need to delete entries? Does it make sense to fetch and report the state for investigating errors at the end of test runs?

Would it make sense to introduce a CLI flag to keep state after a test run for investigation?

Reply:
@urso Good point. We need to delete entries at the beginning of the test to remove all old SQS messages. But these integration tests are not running by CI currently and we are moving to use terraform instead. I will remove this method.